### PR TITLE
[snmp] make use of openHAB network configuration

### PR DESCRIPTION
**Pull request by lsiepel**
Fixes https://github.com/openhab/openhab-addons/issues/17550#issuecomment-2453582050

Confirmed here https://community.openhab.org/t/snmp-seems-not-to-work-as-expected/157319/28

### CODE REVIEW

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java [hunk 1]
```diff
@@ -64,35 +70,42 @@
...
         SnmpServiceConfiguration snmpCfg = new Configuration(config).as(SnmpServiceConfiguration.class);
         try {
             shutdownSnmp();
 
             final DefaultUdpTransportMapping transport;
 
             if (snmpCfg.port > 0) {
-                transport = new DefaultUdpTransportMapping(new UdpAddress(snmpCfg.port), true);
+                InetAddress inetAddress = InetAddress.getByName(networkAddressService.getPrimaryIpv4HostAddress());
+                inetAddress = inetAddress != null ? inetAddress : InetAddress.getLocalHost();
```
Comment (lsiepel) [inline bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:107]: If the network configuration is not set, it falls back to the same behavior as currently is used when the UdpAddress is only provided with a port.

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java [hunk 2]
```diff
@@ -64,35 +70,42 @@
...
+    public SnmpServiceImpl(Map<String, Object> config, @Reference NetworkAddressService networkAddressService) {
         addProtocols();
         OctetString localEngineId = new OctetString(MPv3.createLocalEngineID());
         USM usm = new USM(SecurityProtocols.getInstance(), localEngineId, 0);
         SecurityModels.getInstance().addSecurityModel(usm);
-
+        this.networkAddressService = networkAddressService;
+        networkAddressService.addNetworkAddressChangeListener(this);
         modified(config);
+        this.config = config;
```
Comment (J-N-K) [inline bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:93]: This should not be necessary when `modified` handles the configuration.

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java [hunk 3]
```diff
@@ -109,7 +122,7 @@ protected void modified(Map<String, Object> config) {
             this.snmp = snmp;
             this.transport = transport;
 
-            logger.debug("initialized SNMP at {}", transport.getAddress());
+            logger.info("initialized SNMP at {}", transport.getAddress());
```
Comment (J-N-K) [inline bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:125]:
```suggestion
            logger.info("Initialized SNMP service at {}", transport.getAddress());
```

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java [hunk 4]
```diff
@@ -121,7 +134,7 @@ public void deactivate() {
         try {
             shutdownSnmp();
         } catch (IOException e) {
-            logger.info("could not end SNMP: {}", e.getMessage());
+            logger.warn("could not end SNMP: {}", e.getMessage());
```
Comment (J-N-K) [inline bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:137]:
```suggestion
            logger.warn("Could not end SNMP service: {}", e.getMessage());
```

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java (hunk 1 above)
Comment (J-N-K) [inline bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:107]:
```suggestion
                InetAddress inetAddress = Objects.requireNonNullElse(InetAddress.getByName(networkAddressService.getPrimaryIpv4HostAddress(), InetAddress.getLocalHost());
```

**Review by J-N-K — CHANGES_REQUESTED** [review]
Just some style comments

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:93]: Fixed.

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java (hunk 3 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:125]: Fixed

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java (hunk 4 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:137]: Fixed

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java:107]: Fixed

**Review by J-N-K — APPROVED** [review]
Thanks!

**Pull request merged**